### PR TITLE
support attention mask in user's dataset

### PR DESCRIPTION
**Pull request by wenhuach21**

### CODE REVIEW

**Comment by wenhuach21** [comment]
This PR is not very robust since the `attention_mask `is not incorporated into the forward pass since different models may handle it differently. Therefore, we should warn users when their input data includes padding. As a workaround, they can use a batch size of 1 instead.

**Review by WeiweiZhang1 — APPROVED** [review]
